src/sortStudents.ts

```diff
@@ -1,16 +1,68 @@
-
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): object[] {
+  const copyOfStudents: object[] = students.map(
+    (student: object): object => {
+      return { ...student };
+    },
+  );
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      copyOfStudents.sort((studentA: object, stundentB: object) => (
```

Comment:
```suggestion
      copyOfStudents.sort((studentA: Student, stundentB: Student) => (
```
we must type not just as an object, but as an interface. do it in the code below

Reply:
I understand but everytime I wrote that i had this problem 
![image](https://user-images.githubusercontent.com/109978413/193788346-09d6748d-8a25-431b-8266-88546e8f89f7.png)

Reply:
it can be because you subtract: true - false, false - true, etc...
you need to convert that values into numbers

Reply:
I changed it but still problem remain and even for a different cases, but anyway I changed all for Student so...
![image](https://user-images.githubusercontent.com/109978413/193795092-924d9070-685a-4db5-9667-6f00653fefd9.png)

Reply:
fixed this